src/community/wmts-multi-dimensional/src/main/java/org/geoserver/gwc/wmts/dimensions/CoverageDimensionsReader.java

```diff
@@ -162,6 +162,22 @@ public String getGeometryAttributeName() {
                 throw new RuntimeException("Error reading domain values.", exception);
             }
         }
+
+        @Override
+        public ReferencedEnvelope getBounds(Filter filter) {
+            try {
+                GranuleSource source = reader.getGranules(reader.getGridCoverageNames()[0], true);
+                Query query = new Query(null, filter);
+                if (filter != null) {
```

Comment:
I'm not sure, but if we provide the filter in the constructor do we need do set it here ?

Reply:
Good catch :)